GitHubBranches.ps1

```diff
@@ -192,11 +536,28 @@ filter Add-GitHubBranchAdditionalProperties
 
         if (-not (Get-GitHubConfiguration -Name DisablePipelineSupport))
         {
-            $elements = Split-GitHubUri -Uri $item.commit.url
+            if ($null -ne $item.url)
+            {
+                $elements = Split-GitHubUri -Uri $item.url
+            }
+            else
+            {
+                $elements = Split-GitHubUri -Uri $item.commit.url
+            }
             $repositoryUrl = Join-GitHubUri @elements
+
             Add-Member -InputObject $item -Name 'RepositoryUrl' -Value $repositoryUrl -MemberType NoteProperty -Force
 
-            Add-Member -InputObject $item -Name 'BranchName' -Value $item.name -MemberType NoteProperty -Force
+            if ($null -ne $item.name)
+            {
+                $branchName = $item.name
+            }
+            else
+            {
+                $branchName = $item.ref -replace ('refs/heads/', '')
+            }
```

Comment:
Similarly, you're accessing `$branchName` below out-of-scope from where it was defined.  Let's simplify this statement and ensure its accessible:

```suggestion
            $branchName = $item.name
            if ($null -eq $branchName)
            {
                $branchName = $item.ref -replace ('refs/heads/', '')
            }
```

Reply:
Horrible code, but life is too short. Done.